src/Microsoft.AspNetCore.Routing/ParameterTransformer.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public abstract class ParameterTransformer : IParameterPolicy
+    {
+        public abstract string Transform(string value);
```

Comment:
I think we need the ability to run this on an inbound value as well don't we?

Reply:
I guess when you consider that, you'd want to have another argument that indicates the direction (or a separate method, which I'd probably prefer).